### PR TITLE
Added calicoctl ability to operate on bgpconfig, felixconfig, and clusterinfo

### PR DESCRIPTION
**Pull request by mgleung**
## Description
Added the resource types and commands for `FelixConfiguration`, `BGPConfiguration`, and `ClusterInformation`.  Also added the tests against the basic operations.

## Todos
- [x] Tests
- [x] Documentation (not needed here)
- [x] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: calicoctl/commands/get.go [hunk 1]
```diff
@@ -70,6 +70,9 @@ Description:
     * networkPolicy
     * globalNetworkPolicy
     * profile
+    * bgpConfiguration
+    * felixConfiguration
+    * clusterInformation
```
Comment (robbrockbank) [inline calicoctl/commands/get.go]: given the size of this list now, I wonder if we should either (a) alphabetize this, or (b) just refer to the documentation.

perhaps you could do a quick poll and see what people think.

or maybe take a look at the kubectl help and see if that specifies the valid resource types.

File: calicoctl/resourcemgr/clusterinfo.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	"github.com/projectcalico/libcalico-go/lib/options"
+)
+
+func init() {
+	registerResource(
+		api.NewClusterInformation(),
+		api.NewClusterInformationList(),
+		false,
+		[]string{"clusterinformation", "clusterinformations", "clusterinfo", "clusterinfos"},
+		[]string{"NAME"},
```
Comment (robbrockbank) [inline calicoctl/resourcemgr/clusterinfo.go]: might as well include all of the fields for this one.

File: calicoctl/resourcemgr/bgpconfig.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	"github.com/projectcalico/libcalico-go/lib/options"
+)
+
+func init() {
+	registerResource(
+		api.NewBGPConfiguration(),
+		api.NewBGPConfigurationList(),
+		false,
+		[]string{"bgpconfiguration", "bgpconfigurations", "bgpconfig", "bgpconfigs"},
+		[]string{"NAME"},
```
Comment (robbrockbank) [inline calicoctl/resourcemgr/bgpconfig.go]: include all of the fields for this one

Comment (robbrockbank) [reply calicoctl/resourcemgr/bgpconfig.go]: you might need to do some magic to handle nil values in the go template below. 

for now, maybe just display nil as a dash...?

**Comment by robbrockbank** [comment]
could you remove the config command too since that's no longer valid.

File: calicoctl/resourcemgr/bgpconfig.go (hunk 3 above)
Comment (mgleung) [reply calicoctl/resourcemgr/bgpconfig.go]: I have displayed `<nil>` as `----`.
Ex:
```
NAME         LOGSEVERITY   MESHENABLED   DEFAULTASN
default      Info          false         6511
testConfig   Info          ----          ----
```

File: calicoctl/resourcemgr/bgpconfig.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+		api.NewBGPConfiguration(),
+		api.NewBGPConfigurationList(),
+		false,
+		[]string{"bgpconfiguration", "bgpconfigurations", "bgpconfig", "bgpconfigs"},
+		[]string{"NAME", "LOGSEVERITY", "MESHENABLED", "DEFAULTASN"},
+		[]string{"NAME", "LOGSEVERITY", "MESHENABLED", "DEFAULTASN"},
+		map[string]string{
+			"NAME":        "{{.ObjectMeta.Name}}",
+			"LOGSEVERITY": "{{.Spec.LogSeverityScreen}}",
+			"MESHENABLED": "{{if .Spec.NodeToNodeMeshEnabled}}{{.Spec.NodeToNodeMeshEnabled}}{{ else }}----{{ end }}",
```
Comment (robbrockbank) [inline calicoctl/resourcemgr/bgpconfig.go]: A single dash here would be fine.

**Pull request merged**